packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/credential_store_state_machine.dart

```diff
@@ -208,6 +209,14 @@ class CredentialStoreStateMachine extends StateMachine<CredentialStoreEvent,
           expiration,
         );
       }
+      final signInDetailsStr = await _secureStorage.read(
```

Comment:
apiBased and hostedUi do not store the signInDetails json string and instead store specified fields. would it be better to do the same for federated?

Reply:
Good question. I can change it to match the others. There is a tradeoff between time spent reading values and time spent decoding values - I think consistency may be a bigger win though.